### PR TITLE
Make some members protected in WebotsNode.hpp

### PR DESCRIPTION
**Pull request by dzywater**
Make some members protected in WebotsNode.hpp for users' further development and ideas' try
  mRobot 
  mRobotName 
  mRobotDescription 
  mClient s
  setAnotherNodeParameter

### CODE REVIEW

File: webots_ros2_driver/include/webots_ros2_driver/WebotsNode.hpp [hunk 1]
```diff
@@ -44,19 +44,21 @@ namespace webots_ros2_driver
     webots::Supervisor *robot() { return mRobot; }
     std::string urdf() const { return mRobotDescription; };
 
+  protected:
+      webots::Supervisor* mRobot;
```
Comment (lukicdarkoo) [inline webots_ros2_driver/include/webots_ros2_driver/WebotsNode.hpp:48]: This is accessible through the `robot()` getter.

File: webots_ros2_driver/include/webots_ros2_driver/WebotsNode.hpp [hunk 2]
```diff
@@ -44,19 +44,21 @@ namespace webots_ros2_driver
...
+      std::string mRobotName;
+      std::string mRobotDescription;
+
```
Comment (lukicdarkoo) [inline webots_ros2_driver/include/webots_ros2_driver/WebotsNode.hpp:51]: This is accessible through the `urdf()` getter.

**Review by lukicdarkoo — COMMENTED** [review]
Exposing more of API means more maintenance and less flexibility on our side. Could you please tell me your use case?

**Comment by dzywater** [comment]
Oh, sorry, I ignored them suddenly, robot() and urdf() are enough.

**Comment by lukicdarkoo** [comment]
Could you please propose changes to:
https://github.com/cyberbotics/webots_ros2/wiki/Tutorial-Creating-a-Custom-Plugin
so the others are also aware of the methods?

**Comment by lukicdarkoo** [comment]
The parameter name is the same, but the node name is not. To change a parameter, you have to specify a node name as well.

I wouldn't rename it because we eventually want to support full URDF specification:
- https://discourse.ros.org/t/urdf-with-webots/19665
- https://github.com/cyberbotics/urdf2webots/issues/112

**Comment by dzywater** [comment]
> The parameter name is the same, but the node name is not. To change a parameter, you have to specify a node name as well. Therefore, there are no conflicts.
> 
> I wouldn't rename it because we want to implement full URDF support:
> 
>     * https://discourse.ros.org/t/urdf-with-webots/19665
> 
>     * [Question: why not introduce a <webots> tag? urdf2webots#112](https://github.com/cyberbotics/urdf2webots/issues/112)

Wow! That's a good news! 
Which version will plan to implement full URDF support?

**Comment by lukicdarkoo** [comment]
> Which version will plan to implement full URDF support?

We have some ideas on how to implement it, but we haven't settled on anything. Once we do a detailed plan we will be able to estimate the time of release.

**Comment by dzywater** [comment]
> > Which version will plan to implement full URDF support?
> 
> We have some ideas on how to implement it, but we haven't settled on anything. Once we do a detailed plan we will be able to estimate the time of release.

Currently, how could we get webots robot's urdf (Robot::getUrdf) in other ros nodes?

**Comment by lukicdarkoo** [comment]
You read the `robot_description` param. Alternatively, you can make [a simple Python plugin](https://github.com/cyberbotics/webots_ros2/wiki/Tutorial-Creating-a-Custom-Python-Plugin) that publishes a latched topic (or service) with the description.

**Pull request closed without merging**